tests/text_input_v2_with_input_method_v1.cpp

```diff
@@ -0,0 +1,285 @@
+    zwp_input_method_context_v1_delete_surrounding_text(*input_method_context, index, length);
+    zwp_input_method_context_v1_commit_string(*input_method_context, input_method_context->serial, text);
+    input_client.roundtrip();
+    app_client.roundtrip();
+}
+
+TEST_F(TextInputV2WithInputMethodV1Test, input_method_can_send_keysym)
+{
+    uint32_t time = 0;
+    uint32_t sym = 65;
+    uint32_t state = 1;
+    uint32_t modifiers = 0;
+
+    enable_text_input();
+
+    EXPECT_CALL(text_input, keysym(time, sym, state, modifiers));
+    zwp_input_method_context_v1_keysym(
+        *input_method_context,
+        input_method_context->serial,
+        time, sym, state, modifiers);
+    input_client.roundtrip();
+    app_client.roundtrip();
+}
+
+TEST_F(TextInputV2WithInputMethodV1Test, input_method_can_set_preedit_string)
+{
+    auto const preedit_text = "some text";
+    auto const preedit_commit = "some fallback text";
+
+    enable_text_input();
+
+    EXPECT_CALL(text_input, preedit_string(preedit_text, preedit_commit));
+    zwp_input_method_context_v1_preedit_string(
+        *input_method_context,
+        input_method_context->serial,
+        preedit_text,
+        preedit_commit);
+    input_client.roundtrip();
+    app_client.roundtrip();
+}
+
+TEST_F(TextInputV2WithInputMethodV1Test, input_method_can_set_preedit_style)
+{
+    auto const preedit_text = "some text";
+    auto const preedit_commit = "some fallback text";
+    auto const index = 0;
+    auto const length = 3;
+    auto const style = 1;
+
+    enable_text_input();
+
+    EXPECT_CALL(text_input, predit_styling(index, length, style));
+    zwp_input_method_context_v1_preedit_styling(
+        *input_method_context,
+        index,
+        length,
+        style);
+    zwp_input_method_context_v1_preedit_string(
+        *input_method_context,
+        input_method_context->serial,
+        preedit_text,
+        preedit_commit);
+    input_client.roundtrip();
+    app_client.roundtrip();
+}
+
+TEST_F(TextInputV2WithInputMethodV1Test, input_method_can_set_preedit_cursor)
+{
+    auto const preedit_text = "some text";
+    auto const preedit_commit = "some fallback text";
+    auto const index = 3;
+
+    enable_text_input();
+
+    EXPECT_CALL(text_input, preedit_cursor(index));
+    zwp_input_method_context_v1_preedit_cursor(
+        *input_method_context,
+        index);
+    zwp_input_method_context_v1_preedit_string(
+        *input_method_context,
+        input_method_context->serial,
+        preedit_text,
+        preedit_commit);
+    input_client.roundtrip();
+    app_client.roundtrip();
+}
+
+TEST_F(TextInputV2WithInputMethodV1Test, input_method_can_set_modifiers_map)
+{
+    // Note: This example data was taken from the maliit-keyboard example
+    auto const text = "hello";
+    const size_t data_length = 33;
+    wl_array map;
+    wl_array_init(&map);
+    const char** data = static_cast<const char**>(wl_array_add(&map, sizeof(char) * data_length));
```

Comment:
Style nit: You can probably declare this as `auto data`, as you're explicitly casting on the rhs?